src/features/editor/hooks/useAttachments.ts

```
@@ -0,0 +1,201 @@
+const generateId = () => `att_${Date.now()}_${++attachmentIdCounter}`;
+
+/**
+ * 첨부 파일 관리 훅
+ * - 파일 추가/삭제
+ * - 캔버스 이미지 추가
+ * - 파일 탐색기 열기
+ */
+export const useAttachments = (): UseAttachmentsReturn => {
+  const [attachments, setAttachments] = useState<Attachment[]>([]);
+  const attachmentsRef = useRef(attachments);
+  attachmentsRef.current = attachments;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/features/editor/hooks/useAttachments.ts:60:3
  58 |   const [attachments, setAttachments] = useState<Attachment[]>([]);
  59 |   const attachmentsRef = useRef(attachments);
> 60 |   attachmentsRef.current = attachments;
     |   ^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  61 |   const [isDragOver, setIsDragOver] = useState(false);
  62 |   const fileInputRef = useRef<HTMLInputElement>(null);
  63 |   const dragCounterRef = useRef(0);

(react-hooks/refs)